packages/types/src/type/ExtrinsicEra.ts

```diff
@@ -2,33 +2,189 @@
+ * @description
+ * The ImmortalEra for an extrinsic
+ */
+export class ImmortalEra extends U8a {
   constructor (value?: AnyU8a) {
-    super(
-      ExtrinsicEra.decodeExtrinsicEra(value)
-    );
+    super(value);
+
+    assert(this.eq(VALID_IMMORTAL), `IMMORTAL: expected ${VALID_IMMORTAL.toHex()}, found ${this.toHex()}`);
   }
+}
 
-  static decodeExtrinsicEra (value?: AnyU8a): Uint8Array {
-    if (value) {
-      const u8a = u8aToU8a(value);
+export type MortalEraValue = [U8, U8];
 
-      // If we have a zero byte, it is immortal (1 byte in length), otherwise we have
-      // the era details following as another byte
-      return u8a.subarray(0, (u8a[0] === 0) ? 1 : 2);
+interface MortalMethod {
+  current: number;
+  period: number;
+}
+
+/**
+ * @name MortalEra
+ * @description
+ * The MortalEra for an extrinsic, indicating period and phase
+ */
+export class MortalEra extends Tuple {
+  constructor (value?: any) {
+    super({
+      period : U8, phase : U8
+    }, MortalEra.decodeMortalEra(value));
+  }
+
+  private static decodeMortalEra (value: MortalMethod | Uint8Array): MortalEraValue {
+    if (isU8a(value)) {
+      const first = u8aToBn(value.subarray(0, 1)).toNumber();
+      let second = u8aToBn(value.subarray(1, 2)).toNumber();
+      const encoded: number = first + (second << 8);
+      const period = 2 << (encoded % (1 << 4));
+      const quantizeFactor = Math.max(period >> 12, 1);
+      let phase = (encoded >> 4) * quantizeFactor;
+      if (period >= 4 && phase < period) {
+        return [new U8(period), new U8(phase)];
+      }
+      throw new Error('Invalid data passed to Mortal era');
+    } else if (isObject(value)) {
+      const { current } = value;
+      const { period } = value;
+      let calPeriod = Math.pow(2, Math.ceil(Math.log2(period)));
+      calPeriod = Math.min(Math.max(calPeriod, 4), 1 << 16);
+      const phase = current % calPeriod;
+      const quantizeFactor = Math.max(calPeriod >> 12, 1);
+      const quantizedPhase = phase / quantizeFactor * quantizeFactor;
+      return [new U8(calPeriod), new U8(quantizedPhase)];
     }
+    throw new Error('Invalid data passed to Mortal era');
+  }
+  /**
+   * @description The period of this Mortal wraps as a [[U8]]
+   */
+  get period (): U8 {
+    return this[0] as U8;
+  }
+
+  /**
+   * @description The phase of this Mortal wraps as a [[U8]]
+   */
+  get phase (): U8 {
+    return this[1] as U8;
+  }
+
+  /**
+   * @description Encodes the value as a Uint8Array as per the parity-codec specifications
+   * @param isBare true when the value has none of the type-specific prefixes (internal)
+   */
+  toU8a (isBare?: boolean): Uint8Array {
+    const period = this.period.toNumber();
+    const phase = this.phase.toNumber();
+    const quantizeFactor = Math.max(period >> 12, 1);
+    const trailingZeros = this.getTrailingZeros(period);
+    const encoded = Math.min(15, Math.max(1, trailingZeros - 1)) + (((phase / quantizeFactor) << 4));
```

Comment:
In this case, it would make sense to have a short comment of how the structure operates here - it makes maintenance and checking the logic easier without having to know the specs off-by-hand. (Additionally, this can then be checked against any future spec changes as well)

Reply:
added in comments section